.eslintrc

```diff
@@ -0,0 +1,62 @@
+{
+  "extends": "standard-with-typescript",
+  "parserOptions": {
+      "project": [
+        "./tsconfig.json",
+        "./tsconfig.configs.json",
+        "./test/tsconfig.json",
+        "./examples/tsconfig.json",
+        "./perf/tsconfig.json"
+      ]
```

Comment:
The shareable config above uses a parser that replaces ESLint's default parser, in order to be able to lint both JavaScript and TypeScript.

All of the linted files, whether they are TypeScript or JavaScript must go through the TypeScript API. Which means they all must be included in a TypeScript "project". The list above includes both real and "fake" TypeScript projects for that purpose.